src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -1123,6 +1150,20 @@ public Collection<Object> createComponents(
             namedXContentRegistry.get()
         );
 
+        final var resourceSharingIndex = ResourceSharingConstants.OPENSEARCH_RESOURCE_SHARING_INDEX;
```

Comment:
Can we also only instantiate these and return them from createComponents if the feature is enabled?

Reply:
This is only used in 1 place, which is already controlled by feature-flag: https://github.com/DarshitChanpura/security/blob/adf96f1d377ea673fbe5a31644fa731ffff85623/src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java#L2169